cloud/types.go

```diff
@@ -172,3 +172,12 @@ type TagsSpec struct {
 	Tags       infrav1.Tags
 	Annotation string
 }
+
+// PrivateZoneSpec defines the specification for a private DNS zone.
+type PrivateZoneSpec struct {
```

Comment:
wdyt about `PrivateDnsZoneSpec`?

Reply:
changed to `PrivateDnsSpec` since this is more than the zone